### PR TITLE
Test for non-default compression codec

### PR DESCRIPTION
**Pull request by FX196**
- Added a new table compressed with gzip to the test bucket.
- Added tests to make sure the client is able to read the table.

### CODE REVIEW

File: python/delta_sharing/tests/test_reader.py [hunk 1]
```diff
@@ -237,3 +237,12 @@ def list_files_in_table(
     expected = reader.to_pandas().iloc[0:0]
 
     pd.testing.assert_frame_equal(pdf, expected)
+
+
+@pytest.mark.skipif(not ENABLE_INTEGRATION, reason=SKIP_MESSAGE)
+def test_load_gzip_table(rest_client: DataSharingRestClient):
```
Comment (zsxwing) [inline python/delta_sharing/tests/test_reader.py:243]: It's better to add a test in `https://github.com/delta-io/delta-sharing/blob/main/python/delta_sharing/tests/test_delta_sharing.py#L112` instead.

**Review by zsxwing — CHANGES_REQUESTED** [review]
Could you also add a test in https://github.com/delta-io/delta-sharing/blob/main/spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSuite.scala#L32 as well?

File: python/delta_sharing/tests/test_reader.py (hunk 1 above)
Comment (FX196) [reply python/delta_sharing/tests/test_reader.py:243]: Sure, I thought I'd put it here because it was more related to the reader.

**Review by zsxwing — APPROVED** [review]
LGTM

**Pull request merged**